blocks/randompruner.go

```diff
@@ -115,6 +134,13 @@ func (pruner *RandomPruner) DeleteBlock(ctx context.Context, cid cid.Cid) error
 	}
 
 	pruner.size -= uint64(blockSize)
+	pruner.allCidsLk.Lock()
+	cs, ok := pruner.allCids[cid]
+	delete(pruner.allCids, cid)
+	if ok {
```

Comment:
```suggestion
	if ok {
		delete(pruner.allCids, cid)
```
I imagine for a big map that a noop `delete` operation is incurring the same search cost as a plain lookup?

Reply:
I guess it likely would?